test/ng/directive/inputSpec.js

```diff
@@ -2819,6 +2819,313 @@ describe('input', function() {
     });
   });
 
+  describe('range', function() {
+
+    var scope;
+
+    beforeEach(function() {
+      scope = $rootScope;
+    });
+
+    it('should reset the model if view is invalid', function() {
+      var inputElm = helper.compileInput('<input type="range" ng-model="age"/>');
+
+      scope.$apply('age = 100');
+      expect(inputElm.val()).toBe('100');
+
+      try {
+        // to allow non-number values, we have to change type so that
+        // the browser which have number validation will not interfere with
+        // this test. IE8 won't allow it hence the catch.
+        inputElm[0].setAttribute('type', 'text');
+      } catch (e) {}
+
+      helper.changeInputValueTo('100X');
```

Comment:
It can happen in IE9. Browsers that don't implement the range should still be able to validate the input correctly  (we mimic the validation rules if no validity is present )